src/upgrades/mongodb_upgrade.py

```diff
@@ -279,14 +281,19 @@ def is_cluster_healthy(self) -> bool:
                 logger.error("Cannot proceed with upgrade. Service mongod is not running")
                 return False
 
-        # unit status should be
-        unit_state = self.charm.process_statuses()
-        if not isinstance(unit_state, ActiveStatus):
+        if not self.charm.status.are_charm_units_all_active_or_waiting_for_upgrade():
             logger.error(
-                "Cannot proceed with upgrade. Unit is not in Active state, in: %s.", unit_state
+                "Cannot proceed with upgrade. Status of charm units do not show active / waiting for upgrade."
             )
             return False
 
+        if self.charm.is_role(Config.Role.CONFIG_SERVER):
+            if not self.charm.status.are_shards_status_active_or_waiting_for_upgrade():
+                logger.error(
+                    "Cannot proceed with upgrade. Status of shard units do not show active / waiting for upgrade."
```

Comment:
So, you will fall in this "if" every time your cluster is all-green as well. If this is a check you do, say, every hook, then you will log this error every hook.